app/uk/gov/hmrc/customs/managesubscription/connectors/HandleResponses.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.customs.managesubscription.connectors
+
+import play.api.libs.json.{JsError, JsSuccess, Json, OFormat, Reads}
+import uk.gov.hmrc.http.HttpResponse
+import play.mvc.Http.Status.INTERNAL_SERVER_ERROR
+import play.api.Logging
+import uk.gov.hmrc.customs.managesubscription.domain.vat.VatCustomerInformation
+
+trait HandleResponses extends Logging {
+
+  def handleResponse[A](response: HttpResponse)(implicit reads: Reads[A]): Either[ResponseError, A] =
+    response.json.validate[A] match {
+      case JsSuccess(a, _) => Right(a)
+      case JsError(_) =>
+        val error = s"Invalid JSON returned: ${response.body}"
```

Comment:
There's no guarantee that this won't have PII, so let's output the error from within JsError instead of the response body.